tiledb/sm/array/array.cc

```diff
@@ -135,47 +146,60 @@ Status Array::open_without_fragments(
           "Cannot open array without fragments; Array does not exist"));
   }
 
-  // Copy the key bytes.
-  RETURN_NOT_OK(
-      encryption_key_->set_key(encryption_type, encryption_key, key_length));
+  /* Note: query_type_ MUST be set before calling set_array_open()
+    because it will be examined by the ConsistencyController. */
```

Comment:
It will be used by the delete fragments API. There will likely be a new ConsistencyController API added that examines the `Array.query_type()` as part of a check on whether or not a fragment may be deleted.